### PR TITLE
Fix - Updates to Expressive text styles

### PR DESCRIPTION
**Pull request by ccedrone**
Issue: # [1156](https://github.com/cengage/react-magma/issues/1156)

## What I did
- Updated the Expressive text variant: font sizing, weights, and line-heights

## Checklist 
- [X] changeset has been added
- [X] Pull request description is descriptive
- [X] I have made corresponding changes to the documentation
- [X] New and existing unit tests pass locally with my changes
- [N/A] I have added tests that prove my fix is effective or that my feature works

## How to test
- In Docs > Design > Typography, ensure the Expressive Body Text Styles are correct
- In Docs > Design > Typography, ensure the Expressive Heading Text Styles are correct

### CODE REVIEW

File: packages/react-magma-dom/src/theme/magma.ts [hunk 1]
```diff
@@ -575,43 +579,44 @@ export const magma = {
...
+      fontWeight: 500,
     },
     headingXLarge: {
-      mobile: typeScale.size10,
-      desktop: typeScale.size12,
-      fontWeight: 400,
+      mobile: typeScale.size09,
+      desktop: typeScale.size11,
+      fontWeight: 600,
+      lineHeight: typeScale.size13,
```
Comment (silvalaura) [inline packages/react-magma-dom/src/theme/magma.ts:590]: Mmmm, this feels wrong. The lineHeight needs to match the typeScale. So `headingXLarge`'s desktop lineHeight needs to be whatever we set `typeScale.size11.lineHeight`. Lets work with Orion to make sure this is the case

File: packages/react-magma-dom/src/theme/magma.ts [hunk 2]
```diff
@@ -369,6 +369,10 @@ const typeScale = {
     lineHeight: '72px',
   },
   size15: {
+    fontSize: '64px',
+    lineHeight: '84px',
+  },
+  size16: {
```
Comment (silvalaura) [inline packages/react-magma-dom/src/theme/magma.ts:375]: We need to add size16 to the TypeScale interface

File: .changeset/feat-expressiveText.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'react-magma-dom': patch
+---
+
+feat: Updates to the expressive font styles for both desktop and mobile.
```
Comment (silvalaura) [inline .changeset/feat-expressiveText.md:5]: patch and feat is contradictory 🙃

File: packages/react-magma-dom/src/components/Typography/index.tsx [hunk 4]
```diff
@@ -378,6 +378,8 @@ export const headingMediumStyles = props => css`
     @media (min-width: ${props.theme.breakpoints.small}px) {
       font-size: ${props.theme.typographyExpressiveVisualStyles.headingMedium
         .desktop.fontSize};
+      font-weight: ${props.theme.typographyExpressiveVisualStyles.headingSmall
```
Comment (silvalaura) [inline packages/react-magma-dom/src/components/Typography/index.tsx:381]: Should this be `props.theme.typographyExpressiveVisualStyles.headingMedium.fontWeight`?

Comment (ccedrone) [reply packages/react-magma-dom/src/components/Typography/index.tsx:381]: Actually desktop and mobile differ for Medium in regards to font-weight, desktop needs 600 where mobile needs 500 which is currently configured here.

File: packages/react-magma-dom/src/theme/magma.ts (hunk 1 above)
Comment (ccedrone) [reply packages/react-magma-dom/src/theme/magma.ts:590]: `size11` now line-heights a whole `64px`!

File: .changeset/feat-expressiveText.md
Comment (silvalaura) [inline .changeset/feat-expressiveText.md:1]: This file is no longer necessary

File: packages/react-magma-dom/src/components/Typography/index.tsx [hunk 5]
```diff
@@ -239,7 +239,7 @@ export const heading2XLargeStyles = props => css`
       font-size: ${props.theme.typographyExpressiveVisualStyles.heading2XLarge
         .desktop.fontSize};
       line-height: ${props.theme.typographyExpressiveVisualStyles.heading2XLarge
-        .desktop.lineHeight};
+        .mobile.lineHeight};
```
Comment (silvalaura) [inline packages/react-magma-dom/src/components/Typography/index.tsx:242]: should this be desktop? mobile styles are above (lines 231-236)

File: packages/react-magma-dom/src/components/Typography/index.tsx (hunk 4 above)
Comment (silvalaura) [reply packages/react-magma-dom/src/components/Typography/index.tsx:381]: @orion-cengage is it absolutely necessary for desktop and mobile to have different font weights?

File: packages/react-magma-dom/src/components/Typography/index.tsx [hunk 6]
```diff
@@ -268,7 +268,7 @@ export const headingXLargeStyles = props => css`
     font-weight: ${props.theme.typographyExpressiveVisualStyles.headingXLarge
       .fontWeight};
     line-height: ${props.theme.typographyExpressiveVisualStyles.headingXLarge
-      .mobile.lineHeight};
+      .desktop.lineHeight};
```
Comment (silvalaura) [inline packages/react-magma-dom/src/components/Typography/index.tsx:271]: Same here, ideally this should be `mobile`

File: packages/react-magma-dom/src/components/Typography/index.tsx (hunk 4 above)
Comment (orion-cengage) [reply packages/react-magma-dom/src/components/Typography/index.tsx:381]: I will verify with Marketing and report back.

**Review by orion-cengage — CHANGES_REQUESTED** [review]
1. We no longer have to override the color of Body/Large under Expressive. It can just be our default font color.

2. Need to update the em values on the docs page (/design-intro/typography/). The px values have changed but not the em values. Here are the correct values:

Desktop
- H 2XL size 64/4 em, LH 84/5.25 em, medium
- H XL size 48/3 em, LH 64/4 em, semibold
- H L size 36/2.25 em, LH 48/3 em, semibold
- H M size 28/1.75 em, LH 40/2.5 em, _coming soon_
- H S size 24/1.5 em, LH 32/2 em, medium
- H XS size 20/1.25 em, LH 32/2 em, medium
- H 2XS size 18/1.125 em, LH 24/1.5 em, medium

Mobile
- H 2XL size 48/3 em, LH 64/4 em, medium
- H XL size 36/2.25 em, LH 48/3 em, semibold
- H L size 28/1.75 em, LH 40/2.5 em, semibold
- H M size 24/1.5 em, LH 32/2 em, _coming soon_
- H S size 20/1.25 em, LH 32/2 em, medium
- H XS size 18/1.125 em, LH 24/1.5 em, medium
- H 2XS size 16/1 em, LH 24/1.5 em, medium

**Review by orion-cengage — APPROVED** [review]

**Pull request merged**